variants/Generic_F401Rx/variant.cpp

```diff
@@ -22,16 +22,9 @@
 extern "C" {
 #endif
 
-// Pin number
+// Digital PinName array
 const PinName digitalPin[] = {
-  PA_0,  // Digital pin 0
-  PA_1,  // Digital pin 1
-  PA_2,  // Digital pin 2
-  PA_3,  // Digital pin 3
-  PA_4,  // Digital pin 4
-  PA_5,  // Digital pin 5
-  PA_6,  // Digital pin 6
-  PA_7,  // Digital pin 7
+  // PA_0..PA_7 are analog pins
```

Comment:
I didn't saw this before but you have to define those pins here else nothing will works.
This array references all digital pins.